webapp/src/components/TableWrapper/index.tsx

```diff
@@ -0,0 +1,243 @@
+import { FC, ChangeEvent, useState, isValidElement, cloneElement, ReactNode, useEffect } from 'react';
+
+import {
+  Divider,
+  Box,
+  FormControl,
+  InputLabel,
+  Card,
+  TablePagination,
+  Select,
+  MenuItem,
+  CardHeader,
+  Button
+} from '@mui/material';
+import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
+import BulkActions from './BulkActions';
+import { StatusEnum, StatusOptions } from 'src/models/status';
+import { addItem, deleteItem, getItems, updateItem } from 'src/services/common.service';
+import { useModal } from 'src/hooks/useModal';
+import useAuth from 'src/hooks/useAuth';
+import { TableEnum } from 'src/models';
+
+interface TableWrapperProps {
+  className?: string;
+  defaultItems: any;
+  url?: string,
+  title?: string,
+  statusOptions?: StatusOptions,
+  children: ReactNode,
+}
+
+export type TableItem = any;
+
+export interface TableMethods {
+  handleDeleteItem: (id: string) => void,
+  handleUpdateItem: (id: string, data: object) => void,
+  handleGetItems: () => void
+}
+
+export interface TableProps<T> {
+  items: T[], 
+  selectedItems: any,
+  handleSelectAllItems: (event: ChangeEvent<HTMLInputElement>) => void, 
+  handleSelectOneItem: (event: ChangeEvent<HTMLInputElement>, itemId: string) => void,
+  selectedSomeItems: any,
+  selectedAllItems: any,
+  handleDeleteItem: (id: string) => any,
+  handleUpdateItem: (id: string, data: object) => any,
```

Comment:
```suggestion
  handleDeleteItem: (id: string) => void,
  handleUpdateItem: (id: string, data: object) => void,
```

si void fonctionne pas `Promise<void>`